Add support for sending audio sources to multiple buses

Grublady opened the pull request:
This PR implements a `sends` parameter on `AudioStreamPlayer`, `AudioStreamPlayer2D`, `AudioStreamPlayer3D`, and `Area2D`, allowing these types to send their audio stream to more than one bus simultaneously.

Resolves https://github.com/godotengine/godot-proposals/issues/1836.
Supersedes #60316.

### Design

The parameter itself is a `Dictionary[StringName, float]`, where the key is the name of the bus, and the value is the send's gain.
This allows different buses to receive a node's stream at varying volume levels.

It is presented using a normal `Dictionary` inspector property, with its hint string configured to show an option picker for the name and a range slider for the gain.

The existing `bus` property keeps its purpose as the primary bus: while the `sends` dictionary can be empty, `bus` must be specified or it will default to "Master" (even if there are sends present).

All sends respect their node's panning and gain.

Sends are not overridden when an `AudioStreamPlayer2D`/`3D` enters an `Area2D`/`3D` with `audio_bus_override` enabled; in these cases, only the bus specified by `bus` will be overridden.

### Limitations

The most notable limitation is that sample playback is not supported in this current implementation. If using sample playback, audio will only be sent to the primary bus as specified by `bus`.

Additionally, sends on `Area3D` have not been implemented due to complexity with the already-existing reverb send parameter.

These might be good to follow up with in the future, although I don't believe they are central enough to preclude this feature.

### Compatibility

All preexisting outward API is unmodified, and care has been taken to preserve existing behavior, so this should not break compatibility.

Grublady committed:
Add support for sending audio sources to multiple buses

Calinou reviewed (commented):
Tested locally, it works as expected.

**Testing project:** [spectrum_pr_104096.zip](https://github.com/user-attachments/files/19457655/spectrum_pr_104096.zip)

![image](https://github.com/user-attachments/assets/e29dfd07-9042-4053-a367-574c26db732e)

Some feedback:

- You can add a property without any key defined, which results in the following:

![image](https://github.com/user-attachments/assets/ae6c1032-8334-40bf-9fad-f57e699fbdf3)

This entry with an empty name will send to Master, so the sound will play twice as loud if it's there.

- The volume for each send can't be set above `0.0 dB`, which I assume is intentional to avoid infinite feedback loops or something of the sort.

rossunger reviewed (approved):
Looks great :) The only thing I would say is it would be important to clarify in the docs that volume_db now function like "Gain", changing the pre-send level... if it wasn't such a hassle I might even suggest renaming "volume_db" to "gain", especially in the inspector... but I think that would be a nightmare :P

Grublady commented:
> You can add a property without any key defined

Fixed.

> The volume for each send can't be set above 0.0 dB

This was an intentional design decision, although there shouldn't be any technical reason we couldn't change it.
It was modeled after sends in Ableton Live, which go up to 0dB. I believe some other DAWs (Logic?) have send gain go up to +6dB. I'm not opposed to tweaking this if that's what we decide.

>clarify in the docs that volume_db now function like "Gain", changing the pre-send level

I've updated the docs with a clarifying sentence.

Grublady commented:
`godot-cpp` check pending godotengine/godot-cpp#1763

Grublady commented:
Updated docs to specify that sends are ignored when using sample playback.

Mickeon reviewed (commented):
I think this is a very interesting and powerful feature, but I'm not sure it fits **AudioStreamPlayer** & co. given its complexity.

If the **AudioServer** allowed for it, this could just as well be an addon. It's a bit unfortunate that all Playbacks have to go through an **AudioStreamPlayer** node and there's almost no other way around it.
